### PR TITLE
Use either glyph or character range depending on API

### PR DESCRIPTION
**Pull request by TomasLinhart**
It is not always true one glyph equals one character (Arabic alphabet uses multiple glyphs per character), and some methods accept glyph range and some character range, so it is essential to use appropriate range. The existing logic was causing an infinite loop inside of UIKit that never finished and frozen the app and eventually consumed all available memory.

These changes make sure the correct range is always being passed in.

### CODE REVIEW

**Comment by rjchatfield** [comment]
Great work!

I would wait for Rajdeep's approval before merging this.

**Comment by rajdeep** [comment]
ya! this is really great. Thanks for catching this issue and applying the fix @TomasLinhart. I am happy to merge - however, just looking at the issue with github not running the Workflow on forked repos 😕

**Review by rajdeep — APPROVED** [review]

**Pull request merged**
